features/permissions/permissions.step-definitions.js

```diff
@@ -0,0 +1,152 @@
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+const ENV = require('../../environment-variables');
+const request = require('request-promise-native');
+const btoa = require('btoa');
+let accessToken = "a"; // a variable to store the access token
+
+Given('I have a valid credentials, for permissions', function () {
```

Comment:
Would it make sense to put the Then() functions right underneath their related When() functions? Makes it easier to see the flow of the tests.